### ISSUE TITLE
Minor: Proper way to detect, if RTTI is availabe for a certain type

### ISSUE THREAD

**Issue by bogeeee** [typescript-rtti/typescript-rtti#109]
Hey guys, 
your library is working very fine so far (extra properties check is also works good now) and i'm heavily working, to bring the next proper restfuncs release, which will hopefully bring in some more users  ;)

What could still be improved, is a check, to see if RTTI is really available for a type / method. At the moment, `reflect(something)` returns a valid ReflectedXXX all the time and the members are just empty. I have to do some very hacky workarounds, to see, if the info is accurate (does the method really have 0 args, or is there just no RTTI enhanced). Also i have to check for methods that were actually declared at parent class level, where the parent class file is from a non another code base. Would be nice, to make this more convenient.

**Comment by rezonant** [comment]
Yeah, this is because it will give you the information that is normally available from the JS runtime via the ReflectedClass et al APIs, which is nice. I am definitely fine adding a property that lets you know what level of introspection is available.
